src/Microsoft.AspNetCore.Mvc.Razor.Host/MvcRazorHost.cs

```diff
@@ -297,6 +315,33 @@ public override RazorParser DecorateRazorParser(RazorParser razorParser, string
             return new MvcRazorParser(razorParser, inheritedChunkTrees, DefaultInheritedChunks, ModelExpressionType);
         }
 
+        public override RazorParser DecorateRazorParser(
+            RazorParser incomingRazorParser,
+            TemplateEngineContext context)
```

Comment:
Unfortunately we have to thread the relative path all the way here. Seemed cleaner to flow the context in entirety rather than reuse the existing method and pass in the relativePath as part of the invocation. Let me know if you'd us to not introduce additional surface area.